### DISCUSSION TITLE
Necesito ayuda con la separación de letras de la fuente

### DISCUSSION THREAD

**Discussion by MaybeDavey** [Kaplas80/TF3#58] [Q&A]
¡Hola, Kaplas! Estoy trabajando con un grupo de personas en una traducción para el resto de juegos de yakuza, utilizando las herramientas creadas por el grupo Brazil Alliance. El problema radica en la notable separación de algunas letras como la "í" en los subtítulos de los juegos (Yakuza 3 al 5), ya que no se puede modificar el espaciado de las letras en este programa. Queríamos pedirte ayuda, aunque sean unas pistas clave que nos des para poder crear una solución; como los juegos tienen básicamente el mismo motor del 0 y el Kiwami, supuse que podrías tener una idea.

Sé que es una petición muy repentina, pero no se me ocurre nadie más a quien pedirle por favor. 😓

**Comment by Kaplas80** [comment]
Hola!

Perdona que no te haya contestado antes, pero no había visto el mensaje. En todos los juegos de Yakuza que he visto, la tabla que define los espacios de los caracteres está dentro del .exe:

```
Yakuza 0 - Offset: 0x00D8F9E0
Yakuza Kiwami - Offset: 0x00CFF560
Yakuza Kiwami 2 - Offset: 0x024CED30
Yakuza 6 - Offset: 0x01EC4A00
```

En todos los casos es un formato parecido: Un array de 256 elementos (uno por cada caracter ISO 8859-1), en el que cada elemento está compuesto de 6 números en coma flotante de 4 bytes cada uno. [Aquí](https://github.com/Kaplas80/TranslationFramework2/blob/39e87ce18026289b9f10d68fd7242c0b9af84f16/src/Plugins/YakuzaGame/Files/Exe/File.cs#L65) puedes ver como hago la lectura.

Cada número representa el espacio que deja antes y después de cada carácter en este orden:

1. TopLeft
2. TopRight
3. MiddleLeft
4. MiddleRight
5. BottomLeft
6. BottomRight

A la hora de dibujar cada carácter en el juego, se hace un calculo entre los valores 'Right' del carácter `i-1` y los valores 'Left' del carácter `i`, así que, cambiando esos valores, podeis cambiar el espacio a la hora de representarlos.

En TF2 también teníá un editor de espacios y una aproximación de cómo se veían en el juego, aunque para el Kiwami 2 cambiaron la forma de calcularlo.

En cuanto a cómo encontrar la tabla dentro de cada ejecutable, no es fácil, suelen estar cerca de referencias al texto `font` y los valores de las tablas son parecidos entre los juegos.

Espero que eso os sirva.
